### PR TITLE
Addressed few todo's

### PR DESCRIPTION
**Pull request by riya-singhal31**
This PR updates the go-ceph version to use github.com/ceph/go-ceph/cephfs and fixed few todo by replacing the string comparison with error code check.

### CODE REVIEW

File: internal/cephfs/core/metadata.go [hunk 1]
```diff
@@ -19,8 +19,8 @@ package core
 import (
 	"errors"
 	"fmt"
-	"strings"
 
+	libceph "github.com/ceph/go-ceph/cephfs"
```
Comment (nixpanic) [inline internal/cephfs/core/metadata.go]: consider renaming to `libcephfs`, as `libceph` could be for anything (but does not exist)

Comment (riya-singhal31) [reply internal/cephfs/core/metadata.go]: Ah, yes. Missed it. 
Thanks!

**Review by nixpanic — DISMISSED** [review]

File: deploy/cephcsi/image/Dockerfile [hunk 2]
```diff
@@ -38,6 +38,7 @@ RUN ${GOROOT}/bin/go version && ${GOROOT}/bin/go env
 
 RUN dnf -y install --nodocs \
 	librados-devel librbd-devel \
+    librados-devel libcephfs-devel \
```
Comment (nixpanic) [inline deploy/cephcsi/image/Dockerfile]: `librados-devel` is now listed 2x. Just add `libcephfs-devel` on the same line as `librbd-devel`

File: scripts/Dockerfile.devel [hunk 3]
```diff
@@ -29,6 +29,7 @@ RUN dnf -y install \
 	gcc \
 	librados-devel \
 	librbd-devel \
+    libcephfs-devel \
```
Comment (nixpanic) [inline scripts/Dockerfile.devel]: Can you check the indention?

File: scripts/Dockerfile.test [hunk 4]
```diff
@@ -33,6 +33,7 @@ RUN source /build.env \
 	findutils \
 	librados-devel \
 	librbd-devel \
+    libcephfs-devel \
```
Comment (nixpanic) [inline scripts/Dockerfile.test]: and here as well?

File: scripts/Dockerfile.devel (hunk 3 above)
Comment (riya-singhal31) [reply scripts/Dockerfile.devel]: Locally the earlier one was fine, adding an extra tab makes it properly indent here. So added now.

File: deploy/cephcsi/image/Dockerfile (hunk 2 above)
Comment (riya-singhal31) [reply deploy/cephcsi/image/Dockerfile]: Updated, thanks Niels!

**Review by nixpanic — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

**Comment by nixpanic** [comment]
@mergifyio queue

**Pull request merged**